addons/docs/src/blocks/Source.tsx

```diff
@@ -75,8 +75,12 @@ export const getSourceProps = (
       .map(sourceId => {
         const data = storyStore.fromId(sourceId);
         if (data && data.parameters) {
-          const { storySource } = data.parameters;
-          return storySource && extract(sourceId, storySource);
+          const {
+            storySource,
+            docs: { formatSource },
+          } = data.parameters;
+          const extracted = storySource && extract(sourceId, storySource);
+          return formatSource ? formatSource(extracted, sourceId) : extracted;
```

Comment:
I thought you were planning on making this a parameter enhancer?

Reply:
Maybe our wires got crossed. That would make the parameter enhancer API public, and I don't think we're ready for that?

Reply:
I suppose I could implement this existing API with a parameter enhancer under the hood. Is that what you were thinking?

Reply:
Yeah, that's what I thought you said you were going to do.

I'm not super worried about it right now but it seems better that the user could "inspect" the parameters and see how their `formatSource` function had turned the input parameters into a snippet.